### ISSUE TITLE
Tests and Commands do not have access to common JS modules

### ISSUE THREAD

**Issue by thejdeep** [cypress-io/cypress#6966]
### Current behavior:

It is not possible to share Modules between custom commands and tests.

### Desired behavior:

Variables set into modules from custom commands should be able to be read from tests.

### Test code to reproduce

Is there a way to share JS modules between custom commands and Cypress tests ? Right now, when I try to share a variable defined in a JS module, set the value of the variable in the custom command and try to read it in a test, I get undefined ?
For example,

**jsmodule.js**
```js
export const A = {
a: null,

get a() {
    return this.a;
},

set a(a) {
    this.a = a;
},
};
```

**commands.js**
```js
import {A} from './jsmodule.js';
Cypress.Commands.add('testing', () => {
A.a = 5;
});

```
**Spec file**
```js
import {A} from './jsmodule.js'
beforeEach(function() {
cy.testing();
});
it('print',() => {
console.log(A.a);//undefined !!!
});

```

### Versions
3.8.0
Electron

**Comment by thejdeep** [comment]
I am seeing that another user has the same issue reported as #6779

**Comment by thejdeep** [comment]
@bahmutov Any reference to which direction I should be following ?

**Comment by jennifer-shehane** [comment]
I'm actually getting Max call stack exceeded error when running this. Where is your shared `jsmodule.js` actually within the directory structure? 

I see that you are using an older version of Cypress. Please update to the [current version of Cypress](https://on.cypress.io/changelog) also.

<img width="505" alt="Screen Shot 2020-04-10 at 2 58 06 PM" src="https://user-images.githubusercontent.com/1271364/78976203-b5b44a80-7b3b-11ea-8679-6924ebdce136.png">

**Comment by thejdeep** [comment]
@jennifer-shehane I forked the `cypress-test-tiny` repository and have a commit that demonstrates the issue https://github.com/thejdeep/cypress-test-tiny/commit/11bb4d521162b33e507b2efe6569b8416866f07f . Thanks a lot for willing to help out.

**Comment by thejdeep** [comment]
@jennifer-shehane Any updates on this, please ? We have a release blocked on this.

**Comment by thejdeep** [comment]
@jennifer-shehane Nudging to see if I can receive any help on this. Thank you.

**Comment by jennifer-shehane** [comment]
Code to reproduce:

`support/index.js`

```js
import { Username } from './globals';

Cypress.Commands.add('testAsync', function () {
  // Setting the global variable here
  Username.username = 'John Doe';
  return return cy.wrap(Username.username).should('eq', 'John Doe')
}); 
```

`support/globals.js`

```js
export const Username = {
  currentUsername: 'Test',

  get username() {
    debugger
    return this.currentUsername;
  },

  set username(user) {
    this.currentUsername = user;
  }
}; 
```

`spec.js`

```js
import { Username } from '../support/globals'

it('works', () => {
  cy.testAsync().then(() => {
    // ISSUE : Should be John Doe since I am setting the global inside the command but
    // instead prints Test which is the initial value of the global variable
    cy.wrap(Username.username).should('eq', 'John Doe')
  })
})
```

<img width="433" alt="Screen Shot 2020-05-04 at 3 47 02 PM" src="https://user-images.githubusercontent.com/1271364/80951785-8634ed00-8e1e-11ea-8ff4-e62f2d4f5d58.png">

**Comment by bahmutov** [comment]
Here is what is going on, and I do not consider this a bug, just the way bundling happens.

You can find the supporting code in https://github.com/cypress-io/cypress-test-tiny/commit/08172373ff30d0e78544cc240c0a88155ecf035f

Anything bundled from the support file gets its own bundle. Anything imported from the spec file gets into _second_ bundle. These two bundles are served for the spec

<img width="1440" alt="Screen Shot 2020-05-05 at 12 16 30 PM" src="https://user-images.githubusercontent.com/2212006/81089719-afab5280-8eca-11ea-8212-41ec0269dd51.png">

If you open the `support` bundle you will see the `Username` object from "globals"

<img width="1440" alt="Screen Shot 2020-05-05 at 12 16 44 PM" src="https://user-images.githubusercontent.com/2212006/81089788-c81b6d00-8eca-11ea-8b2c-1e7ff9f6abb6.png">

If you open the spec bundle you will ALSO see the `Username` object` from "globals"

<img width="1440" alt="Screen Shot 2020-05-05 at 12 17 01 PM" src="https://user-images.githubusercontent.com/2212006/81089828-d79ab600-8eca-11ea-9e0a-2d1584deca61.png">

Each bundle has its own copy of "Username" imported from the file, changing a property in one does not change it in the other.

If you want to share data and object references between the support bundle nd the spec bundle, do it through the `window` object.

```js
// support file
import { Username } from './globals'

window.Username = Username

Cypress.Commands.add('printName', () => {
  console.log('command globals name: %s', Username.name)
})
```

```js
// spec file
const Username = window.Username

describe('globals', () => {
  it('shows username', () => {
    console.log('in spec username is "%s"', Username.name)
    console.log('now changed it to "foo"')
    Username.name = 'foo'
    console.log('in spec username is "%s"', Username.name)

    cy.printName()
  })
})
```

<img width="646" alt="Screen Shot 2020-05-05 at 12 23 51 PM" src="https://user-images.githubusercontent.com/2212006/81090155-4d068680-8ecb-11ea-8813-8ebe8ba376c5.png">

**Comment by gorru** [comment]
I'm not sure if I can write here. 
BTW... I'm trying to make a custom reporter with metadata added through custom methods called from tests.
I saw the allure test report which use an Allure object shared between the test and the reporter but I can't reproduce his behaviour. I also tried to create a plugin according to the post opening this thread but it doesn't work.
It seems that the global object of the reporter, the plugin and test execution are different so it's not possibile to share data.
Is this correct? There is a way to achieve data sharing between these items?
Thanks in advance
